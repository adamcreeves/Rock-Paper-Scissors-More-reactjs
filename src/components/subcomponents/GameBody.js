import React, { useState } from "react";
import { availableOptions, handleButtonPress } from "../../utils/Utils";
import ResponseMessage from "./ResponseMessage";
import RPSButton from "./RPSButton";
import ScoreBoard from "./ScoreBoard";
import "../../styles/Main.css";

function GameBody({ player1, player2, resetPlayers }) {
  const [player1Score, setPlayer1Score] = useState(0);
  const [player2Score, setPlayer2Score] = useState(0);
  const [player1Throw, setPlayer1Throw] = useState(3);
  const [player2Throw, setPlayer2Throw] = useState(3);
  const [winnerMessage, setWinnerMessage] = useState("");
  const rockPressed = () => {
    setPlayer1Throw(0);
    handleButtonPress(
      availableOptions[0],
      player1,
      player1Score,
      setPlayer1Score,
      player2,
      player2Score,
      setPlayer2Score,
      setPlayer2Throw,
      setWinnerMessage
    );
  };
  const paperPressed = () => {
    setPlayer1Throw(1);
    handleButtonPress(
      availableOptions[1],
      player1,
      player1Score,
      setPlayer1Score,
      player2,
      player2Score,
      setPlayer2Score,
      setPlayer2Throw,
      setWinnerMessage
    );
  };
  const scissorsPressed = () => {
    setPlayer1Throw(2);
    handleButtonPress(
      availableOptions[2],
      player1,
      player1Score,
      setPlayer1Score,
      player2,
      player2Score,
      setPlayer2Score,
      setPlayer2Throw,
      setWinnerMessage
    );
  };
  const resetPressed = () => {
    setPlayer1Score(0);
    setPlayer2Score(0);
    setWinnerMessage("");
    setPlayer1Throw(3);
    setPlayer2Throw(3);
  };
  const newGamePressed = () => resetPlayers();
  const isNewGame = player1Throw === 3;
  return (
    <div className="gameBody">
      <ScoreBoard
        player1={player1}
        player1Score={player1Score}
        player2={player2}
        player2Score={player2Score}
      />
      <div className="body__buttonRow">
        <RPSButton
          buttonFunc={rockPressed}
          label={"Rock"}
          buttonClass={"button"}
        />
        <RPSButton
          buttonFunc={paperPressed}
          label={"Paper"}
          buttonClass={"button"}
        />
        <RPSButton
          buttonFunc={scissorsPressed}
          label={"Scissors"}
          buttonClass={"button"}
        />
      </div>
      <ResponseMessage
        player1={player1}
        player2={player2}
        playersThrow={player1Throw}
        opponentsThrow={player2Throw}
        winnerMessage={winnerMessage}
      />
      {!isNewGame && (
        <div className="body__gameButton">
          <RPSButton
            buttonFunc={resetPressed}
            label={"Reset Game"}
            buttonClass={"gameButton"}
          />
          <RPSButton
            buttonFunc={newGamePressed}
            label={"New Game"}
            buttonClass={"gameButton"}
          />
        </div>
      )}
    </div>
  );
}

export default GameBody;
