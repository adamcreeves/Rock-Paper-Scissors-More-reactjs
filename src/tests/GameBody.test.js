import React from "react";
import { render } from "@testing-library/react";
import GameBody from "../components/subcomponents/GameBody";

describe("Test GameBody", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<GameBody />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render GameBody and not be null", () => {
    const { asFragment } = render(<GameBody />);
    expect(asFragment()).not.toBeNull();
  });
});
