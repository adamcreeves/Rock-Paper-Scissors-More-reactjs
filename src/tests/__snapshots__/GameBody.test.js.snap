// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GameBody should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="gameBody"
  >
    <div
      class="gameBody__gameButtons"
    >
      <button
        class="gameBody__gameButton"
      >
        New Players
      </button>
    </div>
    <div
      class="scoreBoard"
    >
      <div
        class="scoreBoard__title"
        data-testid="title-component"
      >
        Score:
      </div>
      <div
        class="scoreBoard__body"
      >
        <div
          class="scoreBoard__row"
        >
          <div
            class="scoreBoard__label"
          >
            undefined: 
          </div>
          <div
            class="scoreBoard__score"
          >
            0
          </div>
        </div>
        <div
          class="scoreBoard__row"
        >
          <div
            class="scoreBoard__label"
          >
            undefined: 
          </div>
          <div
            class="scoreBoard__score"
          >
            0
          </div>
        </div>
      </div>
    </div>
    <div
      class="gameBody__buttonsAll"
    >
      <div
        class="gameBody__buttonRow"
      >
        <button
          class="gameBody__button"
        >
          Fire
        </button>
        <button
          class="gameBody__button"
        >
          Glass
        </button>
        <button
          class="gameBody__button"
        >
          Dynamite
        </button>
      </div>
      <div
        class="gameBody__buttonRow"
      >
        <button
          class="gameBody__button"
        >
          Rock
        </button>
        <button
          class="gameBody__button"
        >
          Paper
        </button>
        <button
          class="gameBody__button"
        >
          Scissors
        </button>
      </div>
      <div
        class="gameBody__buttonRow"
      >
        <button
          class="gameBody__button"
        >
          Putin
        </button>
        <button
          class="gameBody__button"
        >
          Lizard
        </button>
        <button
          class="gameBody__button"
        >
          Spock
        </button>
      </div>
    </div>
    <div>
      <div
        class="responseMessage"
      >
        <div
          class="responseMessage__winner"
        >
          Let's play!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
